manager/orchestrator/taskreaper/task_reaper.go

```diff
@@ -86,6 +86,20 @@ func (tr *TaskReaper) Run(ctx context.Context) {
 		if err != nil {
 			log.G(ctx).WithError(err).Error("failed to find tasks with desired state REMOVE in task reaper init")
 		}
+
+		// Tasks that have been marked for shutdown (likley because of a service update)
+		// but have not progressed beyond TaskStatePending also need to be cleaned up.
+		// However, we want to let these tasks go through the regular history cleanup process
+		// as a part of the select loop below.
+		shutdownTasks, err := store.FindTasks(readTx, store.ByDesiredState(api.TaskStateShutdown))
+		if err != nil {
+			log.G(ctx).WithError(err).Error("failed to find tasks with desired state SHUTDOWN in task reaper init")
+		}
+		for _, t := range shutdownTasks {
+			if t.Status.State < api.TaskStateAssigned {
+				tr.cleanup = append(tr.cleanup, t.ID)
```

Comment:
Would this cleanup right away, if these are added here?  If there are any orphaned or remove tasks, line 126 calls `tr.tick()`, which cleans up all tasks in `tr.cleanup` immediately. 

Perhaps this loop should be moved to below the orphaned and removed tasks check?  Would it make sense to add a check for this to make sure that these tasks are handled through the regular history process?

Reply:
Nice catch. I'll fix this. I think I can also add a unit-test for this.